### PR TITLE
"Edit" entry mode with Delete functionality

### LINKED ISSUE #937: Layout for edit entry view (including delete button) + routing
## Requirements
* Should provide the ability to navigate to a laid-out entry edit view on both iPhone and iPad
* Tapping the back button returns the user to the entry edit view

## Visual Design
**InVision**
Entry Detail - Edit: https://mozilla.invisionapp.com/share/BQEUSO0S2#/289206367_Entry_Detail_-_Edit

**Zeplin**
Entry Detail - Edit: https://app.zeplin.io/project/5a4e9ff8032d0923200a2635/screen/5c6dc07d1380fe4628cb6cc4

### PR DESCRIPTION
**Pull request by joeyg**
This was original PR #981 but has been updated with edit and delete functionality.

Fixes #937

### CODE REVIEW

**Comment by sashei** [comment]
This might be a good candidate for @jhugman given that I wrote most of it :p

File: lockbox-ios/View/ItemDetailView.swift [hunk 1]
```diff
@@ -10,57 +10,13 @@ import CoreServices
...
-extension ItemDetailCellConfiguration: Equatable {
-    static func ==(lhs: ItemDetailCellConfiguration, rhs: ItemDetailCellConfiguration) -> Bool {
-        return lhs.value == rhs.value
-    }
-}
-
 class ItemDetailView: UIViewController {
     internal var presenter: ItemDetailPresenter?
     private var disposeBag = DisposeBag()
+    private var swipeBag = DisposeBag()
```
Comment (jhugman) [inline lockbox-ios/View/ItemDetailView.swift]: 🤔

File: lockbox-ios/View/ItemDetailView.swift [hunk 2]
```diff
@@ -148,62 +154,59 @@ extension ItemDetailView: UIGestureRecognizerDelegate {
...
                     }
 
-                    cell.titleLabel.text = cellConfiguration.title
-                    cell.valueLabel.text = cellConfiguration.value
+                    cell.title.text = cellConfiguration.title
 
-                    cell.valueLabel.textColor = cellConfiguration.valueFontColor
+                    cellConfiguration.value
+                            .drive(cell.textValue.rx.text)
+                            .disposed(by: cell.disposeBag)
```
Comment (jhugman) [inline lockbox-ios/View/ItemDetailView.swift]: :thinking:

File: lockbox-ios/View/ItemDetailView.swift [hunk 3]
```diff
@@ -88,57 +45,106 @@ class ItemDetailView: UIViewController {
...
                     }
-                    .bind(to: presenter.onCancel)
-                    .disposed(by: self.disposeBag)
-            }
+                    .bind(to: self.presenter!.onRightSwipe)
+                    .disposed(by: self.swipeBag)
         } else {
-            self.navigationItem.leftBarButtonItem = nil
+            self.navigationController?.interactivePopGestureRecognizer?.delegate = nil
+            self.swipeBag = DisposeBag()
```
Comment (jhugman) [inline lockbox-ios/View/ItemDetailView.swift]: Should we be clearing the existing one, before simply discarding it?

File: lockbox-ios/View/ItemDetailCellConfiguration.swift [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    }
+}
+
+extension ItemDetailCellConfiguration: Equatable {
+    static func ==(lhs: ItemDetailCellConfiguration, rhs: ItemDetailCellConfiguration) -> Bool {
+        return lhs.title == rhs.title
+    }
+}
+
+typealias ItemDetailSectionModelObserver = ((Observable<[ItemDetailSectionModel]>) -> Disposable)
```
Comment (jhugman) [inline lockbox-ios/View/ItemDetailCellConfiguration.swift]: Nice.

**Pull request merged**